addons/hr_attendance/models/hr_attendance.py

```diff
@@ -83,8 +83,13 @@ def _compute_overtime_hours(self):
                        att.check_in as ad
                   FROM hr_attendance att
              INNER JOIN hr_attendance_overtime ot
-                    ON date_trunc('day',att.check_in) = date_trunc('day', ot.date)
-                    AND date_trunc('day',att.check_out) = date_trunc('day', ot.date)
+                    ON date_trunc('day',CAST(att.check_in
+                                                at time zone 'utc'
+                                                at time zone
+                                                (SELECT calendar.tz FROM resource_calendar calendar
+                                                 INNER JOIN hr_employee employee ON employee.id = att.employee_id
+                                                 WHERE calendar.id = employee.resource_calendar_id)
+                                        as date)) = date_trunc('day', ot.date)
```

Comment:
For efficiency and readability i think we should maybe pre-compute this table with something like this
```sql
WITH employee_timezones AS (
    SELECT
        employee.id AS employee_id,
        calendar.tz AS timezone
    FROM
        hr_employee employee
    INNER JOIN
        resource_calendar calendar ON employee.resource_calendar_id = calendar.id
    WHERE
        employee.id IN %s
)
SELECT
    att.id AS att_id,
    att.worked_hours AS att_wh,
    ot.id AS ot_id,
    ot.duration AS ot_d,
    ot.date AS od,
    att.check_in AS ad
FROM
    hr_attendance att
INNER JOIN
    hr_attendance_overtime ot ON att.employee_id = ot.employee_id
INNER JOIN
    employee_timezones etz ON att.employee_id = etz.employee_id
.....
```

Reply:
I have made the suggested changes to the SQL query and added the test case.

Reply:
@bona-odoo looks good, we'll r+ this after the freeze to not conflict a current attendance task